tensorflow/python/data/kernel_tests/iterator_test.py

```diff
@@ -1016,6 +1018,85 @@ def fn():
     self.evaluate(counter_var.initializer)
     self.assertEqual(self.evaluate(fn()), 10)
 
+  def assert_dataset_placement(self, host_dataset, host_iterator, host_tensor,
+                               device_dataset, device_iterator, device_tensor):
+
+    def assert_host_placement(_obj):
+      try:
+        self.assertIn("CPU:0", _obj)
```

Comment:
You can avoid the try catch by doing:

`self.assertTrue("CPU:0" in host_dataset._variant_tensor.device or "" == host_dataset._variant_tensor.device)`